### PR TITLE
fix: row removing while editing error (#590)

### PR DESCRIPTION
**Pull request by moonchanyong**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

I append fisnishEditing in removeRow method but caused same issue.
because react use batch system for rendering,  it's not surely closed and continue removing Row
so let removeRow work after explicitly render

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by jung-han** [comment]
Can one of the admins verify this patch?

File: src/grid.tsx [hunk 1]
```diff
@@ -1032,6 +1033,14 @@ export default class Grid {
   public removeRow(rowKey: RowKey, options: OptRemoveRow = {}) {
     const { treeColumnName } = this.store.column;
 
+    const editingAddr = this.store.focus.editingAddress;
+    if (editingAddr) {
+      const { rowKey, columnName } = editingAddr;
+      const cellVal = this.getValue(rowKey, columnName) || '';
+      this.finishEditing(rowKey, columnName, cellVal.toString());
+      rerender();
```
Comment (js87zz) [inline src/grid.tsx]: Thank you for your PR. But it's need to be fixed. Check as below :)

1. Calling `rerender` function looks inefficient for removing the editing cell.
* preact render the DOM element async. So you can control the timing using setTimeout function (refer the `resetData` function)

2. move the source code to `removeRow` function in `dispatch/data.ts` for maintaining the source code.

3. Only remove the editing layer and focusing layer when removed rowKey is same as editing cell's rowKey

Comment (moonchanyong) [reply src/grid.tsx]: oh I didn't know renderer running asynchronously. 
thanks for your information. i will push again.

**Comment by moonchanyong** [comment]
this issue solved at d93776a0073eae8883723212008097bda14e57bd.
close this issue.

**Pull request closed without merging**